processor/src/main/java/org/mapstruct/ap/internal/model/NestedTargetPropertyMappingHolder.java

```diff
@@ -180,9 +180,15 @@ public NestedTargetPropertyMappingHolder build() {
                         .groupedBySourceReferences
                         .entrySet() ) {
                         PropertyEntry sourceEntry = entryBySP.getKey();
+                        boolean forceUpdateMethodOrNonNestedReferencesPresent =
+                            forceUpdateMethod || !groupedSourceReferences.nonNested.isEmpty();
+                        // if an update method is forced or there are non nested source references then
+                        // we need to create mapping options for forged methods (i.e. Ignore unmapped target properties
+                        // because it the nonNested properties might contain the target properties)
```

Comment:
I think we need to put examples in the comments. I'm kind of lost what happens here.

Reply:
Again, I completely agree with you. It is complex (maybe even too much). I can put some examples here, or even mention the tests that are testing this